### ISSUE TITLE
Could not load file or assembly 'Microsoft.Extensions.DependencyModel, Version=6.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. The system cannot find the file specified.

### ISSUE THREAD

**Issue by eymahesh-sellamuthu** [natemcmaster/DotNetCorePlugins#246]
We are using this McMaster.NETCore.Plugins in .net 6 project and getting this error while plugin is loading.

**To Reproduce**
1. Using this version of the library : McMaster.NETCore.Plugins 1.4.0
2. Run this code: .NET6
3. With these arguments: "We are passing shared libraries"
4. See error:
Microsoft.Azure.WebJobs.Script.Description.FunctionAssemblyLoadContext 
Could not load file or assembly 'Microsoft.Extensions.DependencyModel, Version=6.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. The system cannot find the file specified.

These two dll's is not loading:
Microsoft.Extensions.DependencyModel Version=6.0.0.0
Microsoft.DotNet.PlatformAbstractions, Version=3.1.6.0

**Expected behavior**
Plugin should load properly without any issue. We found that in AssemblyLoadContextBuilder.cs method PreferDefaultLoadContextAssembly -- assembly = _defaultLoadContext.LoadFromAssemblyName(name) -- is not handled with null exception. This need be handle gracefully.

**Screenshots**

**Additional context**
We need to support .NET 6 also.

**Comment by wingchunninja** [comment]
Hi,

I have moved my project from .netcore2.1 to .net6. I have also updated the package to 1.4.0.

When calling this line, I am also getting the following error:
`PluginLoader pluginLoader = PluginLoader.CreateFromAssemblyFile(pluginPath, typeof(IPlugin).Assembly.GetTypes());`

`Exception thrown: 'System.InvalidOperationException' in System.Private.CoreLib.dll: 'Dependency resolution failed for component ..... with error code -2147450750. Detailed error: Hostpolicy must be initialized and corehost_main must have been called before calling corehost_resolve_component_dependencies.`

Have you experienced something like this as well?
